python_modules/libraries/dagster-dg/dagster_dg/cli/env.py

```diff
@@ -42,3 +46,70 @@ def list_env_command(**global_options: object) -> None:
         table.add_row(key, value)
     console = Console()
     console.print(table)
+
+
+# ########################
+# ##### PULL
+# ########################
+
+
+def _get_local_secrets_for_locations(
+    client: DagsterCloudGraphQLClient, location_names: set[str]
+) -> Mapping[str, Mapping[str, str]]:
+    secrets_by_location = {location_name: {} for location_name in location_names}
+
+    result = client.execute(LOCAL_SECRETS_FILE_QUERY)
+    for secret in result["secretsOrError"]["secrets"]:
```

Comment:
will this log the error if there is an error? or raise a KeyError about secrets not being in the result

Reply:
Updated to add tests for error logging.